etw/native/build.gradle

```diff
@@ -0,0 +1,135 @@
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+plugins {
+    id 'cpp-library'
+    id 'cpp-unit-test'
+    id 'visual-studio'
+}
+
+// TODO honor skipWinNative property
+
+def javaIncludeDir = new File(System.env.'JAVA_HOME', "include")
+def programFilesX86 = System.env.'ProgramFiles(x86)' ?: 'C:/Program Files (x86)'
+
+// TraceLoggingProvider.h in Windows 10 SDK
+def winSdkDir = "$programFilesX86/Windows Kits/10"
+if (System.env.APPINSIGHTS_WIN_SDK_PATH) {
+    winSdkDir = System.env.APPINSIGHTS_WIN_SDK_PATH
+    logger.info "Windows SDK path set with environment variable, APPINSIGHTS_WIN_SDK_PATH"
+}
+def vsToolsDir = "$programFilesX86/Microsoft Visual Studio 14.0"
+if (System.env.APPINSIGHTS_VS_PATH) {
+    vsToolsDir = System.env.APPINSIGHTS_VS_PATH
+    logger.info "Visual Studio Build Tools path set with environment variable, APPINSIGHTS_VS_PATH"
+}
+
+def winSdkLibPath
+if (System.env.APPINSIGHTS_WIN_SDK_LIB_PATH) {
+    winSdkLibPath = System.env.APPINSIGHTS_WIN_SDK_LIB_PATH
+    logger.info "Windows SDK Lib path set with environemnt variable, APPINSIGHTS_WIN_SDK_LIB_PATH=${System.env.APPINSIGHTS_WIN_SDK_LIB_PATH}"
+} else {
+    winSdkLibPath = "$winSdkDir/Lib/10.0.18362.0/um"
+}
+
+logger.info "Windows SDK:  $winSdkDir"
+logger.info "SDK Lib path: $winSdkLibPath"
+logger.info "Visual Studio Build Tools: $vsToolsDir"
+
+toolChains.withType(VisualCpp).configureEach  {
+    // FIXME can this toolchain be addressed directly?
+    it.installDir vsToolsDir
+    it.windowsSdkDir winSdkDir
+}
+
+ext {
+    headersDir = "$projectDir/src/main/headers"
+}
+library {
+    baseName = "applicationinsights-java-etw-provider"
+    linkage = [ Linkage.SHARED ]
+    targetMachines = [ machines.windows.x86, machines.windows.x86_64 ]
+    privateHeaders.from file(headersDir)
+    dependencies {
+        implementation files(file("$winSdkLibPath/x64/advapi32.lib"))

```

Comment:
nit: you can also specify the lib files to link using msvc compiler-specific 'pragma comment lib': https://docs.microsoft.com/en-us/cpp/preprocessor/comment-c-cpp?view=vs-2019
That way if you need to manage other dependencies for your native code, you could add more to source rather than maintaining it in the outside gradle script. You'd also need only one line instead of 4 (you can then omit the dependencies section entirely).

Reply:
Nice! I'll definitely do this.